### PR TITLE
Add SWT text region detector

### PR DESCRIPTION
**Pull request by automata**
Now that we have support to libccv, we can start creating components to other useful feature extractors like [SWT detector for text regions](http://libccv.org/doc/doc-swt/).

### CODE REVIEW

**Comment by automata** [comment]
@jonnor I want to get back to that PR in the future, so I'm reopening it as a WIP.

**Comment by automata** [comment]
@GabiThume can you review, please? In summary:
- since both SCD and SWT detectors are only wrappers for native binaries powered by CCV, both use common code which I moved to the same source file and updated both components to use it
- adds SWT based text region detector provided by CCV lib

File: spec/SWTDetect.coffee [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    @timeout 10000
+    canvas = null
+    img = null
+    beforeEach (done) ->
+      fs.readFile __dirname+'/being-d4.jpg', (err, image) ->
+        if err
+          return done err
+        img = new Image
+        img.src = image
+        canvas = new Canvas img.width, img.height
```
Comment (GabiThume) [inline spec/SWTDetect.coffee:42]: Can you add some safety checks related to `src` and `image`?

File: spec/SWTDetect.coffee [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      before (done) ->
+        grps = []
+        out.on 'begingroup', (grp) ->
+          grps.push grp
+        out.once "data", (data) ->
+          results = data
+          done()
+        ins.beginGroup 'foo'
+        ins.send canvas
+        # ins.endGroup()
```
Comment (GabiThume) [inline spec/SWTDetect.coffee]: Why you are not sending `endGroup`?

File: components/SWTDetect.coffee [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    datatype: 'object'
+    required: false
+
+  noflo.helpers.WirePattern c,
+    in: 'canvas'
+    out: ['out', 'error']
+    forwardGroups: true
+    async: true
+  , (canvas, groups, outPorts, callback) ->
+    utils.writeCanvasTempFile canvas, (err, tmpFile) ->
```
Comment (GabiThume) [inline components/SWTDetect.coffee:30]: Can you write some safety checks about input data here? Something like `canvas?.width?>0`

**Pull request merged**